### PR TITLE
Improve Organization Retrieval for Organization Details Page

### PR DESCRIPTION
**Pull request by jadekeegan**
This PR refactors the Organization Details page to use a route resolver to retrieve the organization and its data rather than subscribing to directly within the component. It also fixes the titleResolver so it uses the organization from the route instead of making its own HTTP request to get the organization name.

# Major Changes
- Updated the Organization Service to remove profile retrieval since the profile isn't actually used in that service.
- Added an Organization Resolver TS file to store the `organizationResolver` ResolveFn.
- Refactored the Organization Details Component to use the `organizationResolver` rather than directly calling the service and subscribing to the returned Observable.
- Changed the `title` in the route to be a child of the main route so it can inherit from the Organization Details Component parent route.
- Updated the `titleResolver` ResolveFn to use the parent data to get the organization.

# Testing
All changes were tested thoroughly on the local server by running `honcho start`.

# Developer Notes
Currently, the `titleResolver` function is in the Organization Details component. To match the other resolve functions, we could move it to its own TS file.

### CODE REVIEW

**Review by briannata — APPROVED** [review]
LGTM! You really cleaned up the code and everything looks good on my end

File: frontend/src/app/organization/organization-editor/organization-editor.component.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@
  * @license MIT
  */
 
-import { Component } from '@angular/core';
+import { Component, inject } from '@angular/core';
```
Comment (atoneyd) [inline frontend/src/app/organization/organization-editor/organization-editor.component.ts:10]: I think this inject import is unused

File: frontend/src/app/organization/organization-editor/organization-editor.component.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ export class OrganizationEditorComponent {
     path: ':slug/edit',
     component: OrganizationEditorComponent,
     title: 'Organization Editor',
-    resolve: { profile: profileResolver }
+    resolve: { profile: profileResolver}
```
Comment (atoneyd) [inline frontend/src/app/organization/organization-editor/organization-editor.component.ts:30]: Looks like a space was accidentally removed after profileResolver

File: frontend/src/app/organization/organization-details/organization-details.component.ts [hunk 3]
```diff
@@ -7,30 +7,19 @@
...
 import { ActivatedRoute, ActivatedRouteSnapshot, ResolveFn, Route } from '@angular/router';
 import { MatIconRegistry } from '@angular/material/icon';
 import { DomSanitizer } from '@angular/platform-browser';
-import { Observable, map } from 'rxjs';
 import { MatSnackBar } from '@angular/material/snack-bar';
 import { profileResolver } from '/workspace/frontend/src/app/profile/profile.resolver';
-import { Organization, OrganizationService } from '../organization.service';
-import { Profile, ProfileService } from '/workspace/frontend/src/app/profile/profile.service';
+import { Organization } from '../organization.service';
+import { Profile } from '/workspace/frontend/src/app/profile/profile.service';
```
Comment (atoneyd) [inline frontend/src/app/organization/organization-details/organization-details.component.ts:17]: Maybe a stupid question but is there a reason for us to be importing Profile/use profile on this page?

**Review by atoneyd — CHANGES_REQUESTED** [review]
Some small nit comments but overall looks good

**Pull request merged**